### PR TITLE
Switch from error to warning for inconsistencies in TUDataset Fingerprint.

### PR DESCRIPTION
**Pull request by Dsantra92**
As mentioned in https://github.com/JuliaML/MLDatasets.jl/issues/195#issuecomment-1367701706 there are inconsistencies in raw Fingerprint dataset. This pr allows the inconsistencies to pass through with a warning.

Future TODO: Add a section in docs how to work these kind of datasets.

Closes: #195

### CODE REVIEW

File: src/datasets/graphs/tudataset.jl [hunk 1]
```diff
@@ -63,7 +63,9 @@ function TUDataset(name; dir=nothing)
     source, target = st[:,1], st[:,2]
 
     graph_indicator = readdlm(joinpath(d, "$(name)_graph_indicator.txt"), Int) |> vec      
-    @assert all(sort(unique(graph_indicator)) .== 1:length(unique(graph_indicator)))
+    if !all(sort(unique(graph_indicator)) .== 1:length(unique(graph_indicator)))
+      @warn "Some graph indicators are not present in graph_indicator.txt. Ordering of graph and graph labels may not be consistent. Base.getindex might produce unexpected behaiour for unaltered data."
```
Comment (CarloLucibello) [inline src/datasets/graphs/tudataset.jl:67]:
```suggestion
      @warn "Some graph indicators are not present in graph_indicator.txt. Ordering of graph and graph labels may not be consistent. Base.getindex might produce unexpected behavior for unaltered data."
```

File: src/datasets/graphs/tudataset.jl [hunk 2]
```diff
@@ -91,6 +93,8 @@ function TUDataset(name; dir=nothing)
                         readdlm(joinpath(d, "$(name)_graph_attributes.txt"), ',', Float32)' |> collect :
                         nothing
 
+    # TODO: maybe introduce consitency in graph labels and attributes if possible
```
Comment (CarloLucibello) [inline src/datasets/graphs/tudataset.jl:96]:
```suggestion
    # TODO: maybe introduce consistency in graph labels and attributes if possible
```

**Review by CarloLucibello — APPROVED** [review]

**Comment by CarloLucibello** [comment]
nice, thanks! Are tests passing locally?

**Comment by Dsantra92** [comment]
Tests for fingerprint running locally.

**Pull request merged**